File: Exemplos/App_eXperience_Java/Elgin_AppExperience_Java-font/app/src/main/java/com/elgin/elginexperience/Services/Pix4/Produto.java
package com.elgin.elginexperience.Services.Pix4;

import com.elgin.elginexperience.R;

public enum Produto {
    ABACAXI("7.00", "P1.jpg", R.drawable.pix4_image_abacaxi),
    BANANA("8.50", "P2.jpg", R.drawable.pix4_image_banana),
    CHOCOLATE("10.00", "P3.jpg", R.drawable.pix4_image_chocolate),
    DETERGENTE("6.00", "P4.jpg", R.drawable.pix4_image_detergente),
    ERVILHA("4.00", "P5.jpg", R.drawable.pix4_image_ervilha),
    FEIJAO("8.00", "P6.jpg", R.drawable.pix4_image_feijao),
    GOIABADA("9.00", "P7.jpg", R.drawable.pix4_image_goiabada),
    HAMBURGUER("14.00", "P8.jpg", R.drawable.pix4_image_hamburguer),
    IOGURTE("11.00", "P9.jpg", R.drawable.pix4_image_iogurte),
    JACA("9.00", "P10.jpg", R.drawable.pix4_image_jaca);

    final String nome; // Nome do produto. Será utilizado na tela do PIX 4.
    final String preco; // Preço do produto. Será utilizado na tela do PIX 4.

    /* O nome do arquivo, este parâmetro é utilizado para o salvamento da imagem do produto (localizadas em res/drawable) dentro do Minipdv. As imagens serão salva com este nome dentro do diretório /Download/.
     * Também é utilizado para o carregamento da imagem dentro do PIX 4, para que seja possível mostrar a imagem no dispositivo é necessário, primeiramente, carregar no dispositivo (com a função PIX4.CarregaImagemDisplay)
     * passando uma String que será utilizada para referênciar a imagem salva na hora do carregamento com (PIX4.ApresentaImagemDisplay()).
     * Exemplo: PIX4.CarregaImagemDisplay("P1.jpg") executada, para que a imagem carregada seja mostrada na tela deve ser passado exatamente a mesma String no carregamento, PIX4.ApresentaImagemDisplay("P1.jpg"). */
    final String fileName;

    final int fileImageResId; // Id do resource que sera carregado como bitmap, salvo como imagem no Minipdv e depois carregado, a partir da referência da localização onde foi salvo no Minipdv, para dentro do PIX 4.

    Produto(String preco, String fileName, int fileImageResId) {
        this.nome = this.name();
        this.preco = preco;
        this.fileName = fileName;
        this.fileImageResId = fileImageResId;
    }

}

File: Exemplos/App_eXperience_XamarinAndroid/Elgin_AppExperience_XamarinAndroid/Pix4/Produto.cs
using System;

namespace M8
{
    public class Produto
    {
        Produto(string nome, string preco, int fileImageResId,  string fileName)
        {
            Nome = nome;
            Preco = preco;
            FileImageResId = fileImageResId;
            FileName = fileName;
        }

        public string Nome { get; private set; }
        public string Preco { get; private set; }
        public int FileImageResId { get; private set; }
        public string FileName { get; private set; }

        public static Produto ABACAXI = new Produto("ABACAXI", "7.00", Resource.Drawable.pix4_image_abacaxi, "P1.jpg");
        public static Produto BANANA = new Produto("BANANA", "7.00", Resource.Drawable.pix4_image_banana, "P2.jpg");
        public static Produto CHOCOLATE = new Produto("CHOCOLATE", "7.00", Resource.Drawable.pix4_image_chocolate, "P3.jpg");
        public static Produto DETERGENTE = new Produto("DETERGENTE", "7.00", Resource.Drawable.pix4_image_detergente, "P4.jpg");
        public static Produto ERVILHA = new Produto("ERVILHA", "7.00", Resource.Drawable.pix4_image_ervilha, "P5.jpg");
        public static Produto FEIJAO = new Produto("FEIJAO", "7.00", Resource.Drawable.pix4_image_feijao, "P6.jpg");
        public static Produto GOIABADA = new Produto("GOIABADA", "7.00", Resource.Drawable.pix4_image_goiabada, "P7.jpg");
        public static Produto HAMBURGUER = new Produto("HAMBURGUER", "7.00", Resource.Drawable.pix4_image_hamburguer, "P8.jpg");
        public static Produto IOGURTE = new Produto("IOGURTE", "7.00", Resource.Drawable.pix4_image_iogurte, "P9.jpg");
        public static Produto JACA = new Produto("JACA", "7.00", Resource.Drawable.pix4_image_jaca, "P10.jpg");

        public static Produto[] values = new Produto[] { ABACAXI, BANANA, CHOCOLATE, DETERGENTE, ERVILHA, FEIJAO, GOIABADA, HAMBURGUER, IOGURTE, JACA };
    }
}